ext/fetch/lib.rs

```diff
@@ -189,6 +191,26 @@ pub struct FetchReturn {
   cancel_handle_rid: Option<ResourceId>,
 }
 
+pub fn get_or_create_client_from_state(
```

Comment:
Let me know if you can think of any better alternatives to having this function. This didn't seem too bad, but at the same time not that great.

Reply:
I think it's fine, communicates intent clearly